utils/synthetic_data_gen/src/synthetic_data_gen.py

```diff
@@ -0,0 +1,330 @@
+        out_file: Optional[str] = None,
+        breakpoint_threshold_amount: Optional[int] = None,
+        min_doc_length: Optional[int] = None,
+    ) -> None:
+        """
+        Generate synthetic dataset in jsonl file for a given list of documents for LLM fine-tuning.
+
+        Parameters:
+        documents (Union[list, str]): A single string document or a list of string documents.
+        amount (Optional[int], optional): The number of question answer pairs to generate per document.
+        include_context (Optional[bool], optional): Whether to include the context in the question.
+        include_thoughts (Optional[bool], optional): Whether to include the rezoning thought in the answer.
+        include_references (Optional[bool], optional): Whether to include the references in the answer.
+        out_file (Optional[str], optional): The path to save the generated question answer pairs in JSONL format.
+        breakpoint_threshold_amount (Optional[int], optional): The threshold for determining the breakpoint
+            for splitting the original document.
+        min_doc_length (Optional[int], optional): The minimum length for a document after splitting.
+
+        Returns:
+            None
+        """
+        if out_file is None:
+            out_file = self.generation_config['output_path']
+        if amount is None:
+            amount = self.generation_config['amount_per_document']
+        if include_context is None:
+            include_context = self.generation_config['include_context']
+        if include_thoughts is None:
+            include_thoughts = self.generation_config['include_thoughts']
+        if include_references is None:
+            include_references = self.generation_config['include_references']
+        if breakpoint_threshold_amount is None:
+            breakpoint_threshold_amount = self.splitting_config['breakpoint_threshold_amount']
+        if min_doc_length is None:
+            min_doc_length = self.splitting_config['min_doc_length']
+
+        if isinstance(documents, str):
+            documents = [documents]
+
+        documents = self.split_documents(
+            documents=documents, breakpoint_threshold_amount=breakpoint_threshold_amount, min_doc_length=min_doc_length
+        )
+        for document in documents:
+            try:
+                qa_pairs = self.generate_qa_pairs(
+                    context=document.page_content,
+                    amount=amount,
+                    include_context=include_context,
+                    include_thoughts=include_thoughts,
+                    include_references=include_references,
+                )
+                lines = self.qa_pairs_to_prompt_completion(qa_pairs)
+                self.update_jsonl(out_file, lines)
+                logging.info(f'Added {amount} qa pairs to {out_file}')
+            except:
+                logging.warning(f'Failed to generate qa pairs for document: "{document}", skipping')
```

Comment:
Should we print the specific error as well?

Reply:
added